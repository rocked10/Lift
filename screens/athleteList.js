import React, { useState, useEffect } from 'react';
import { View, Text, TouchableOpacity, StyleSheet, FlatList, Keyboard, Alert, } from "react-native";
import * as Auth from '../api/auth';
import * as DB from '../api/database';
import { Button, TextInput } from "react-native-paper"
import { set } from 'react-native-reanimated';
import ProfileCard from "../shared/profileCard";

export default function AthleteList() {
    const coachId = Auth.getCurrentUserId();

    const [coachProfile, setCoachProfile] = useState({});

    useEffect(() => {
        DB.getUserProfile(Auth.getCurrentUserId(), setCoachProfile);
    }, []);

    const [searchQuery, setSearchQuery] = useState('')

    const handleAddAthlete = async () => {
        // Keyboard.dismiss()
        const athlete = await DB.findUserId(searchQuery.toLowerCase().trim());
        if (athlete) {
            DB.addAthlete(coachId, athlete.id).then();
            Alert.alert(
                '',
                'Athlete added!',
                [
                    {
                        text: "Ok",
                    }
                ]
            );
            this.textInput.clear();
        } else {
            Alert.alert(
                '',
                'Athlete not found',
                [
                    {
                        text: "Ok",
                    }
                ]
            );
        }
    }

    const FlatListItemSeparator = () => {
        return (
          //Item Separator
          <View style={styles.listItemSeparatorStyle} />
        );
    };

    // why didn't it work when i placed the textinput and button outside??

    return (
        <View style={styles.container}>
            <TextInput placeholder="Enter Athlete's email" onChangeText={query => setSearchQuery(query)} ref={ input => { this.textInput = input }}/>
            <Button onPress={handleAddAthlete}>add</Button>
            <FlatList
                ItemSeparatorComponent={FlatListItemSeparator}
                data={coachProfile.athletes ? Object.values(coachProfile.athletes) : null}
                renderItem={
                    ({ item }) => (
                        // <TouchableOpacity>
                        //     <Text style={styles.item}>{item.id}</Text>
                        // </TouchableOpacity>

                        <ProfileCard title={item.name} subtitle='Athlete' />
                    )}
                keyExtractor={(item, index) => index}
            />
        </View>
    )
}

const styles = StyleSheet.create({ 
    container: {  
        flex: 1,   
    },   
    item: {  
        padding: 10,
        fontSize: 16,  
        height: 40,  
    },
    listItemSeparatorStyle: {
        height: 0.5,
        width: '100%',
        backgroundColor: '#C8C8C8',
    },  
})  