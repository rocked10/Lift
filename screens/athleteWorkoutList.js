import React, { useState, useEffect } from 'react';
import { View, Text, FlatList, TouchableOpacity, StyleSheet } from "react-native";
import { globalStyles } from "../styles/global";
import Card from "../shared/card";
import * as Auth from '../api/auth';
import * as DB from '../api/database'
import { MaterialCommunityIcons, } from "@expo/vector-icons";

export default function AthleteWorkoutList({ route, navigation }) {
    const { workouts, name } = route.params
    let workoutList = [];

    for (const workout in workouts) {
        const athleteWorkout = workouts[workout];
        workoutList.push(athleteWorkout)
    }

    return (
        <View style={globalStyles.container} >

            <View style={{flexDirection: 'row'}}>
                <MaterialCommunityIcons
                    name='arrow-left'
                    size={26}
                    color='black'
                    onPress={() => navigation.goBack()}
                />

                <Text style={globalStyles.titleText}>{name}'s Workouts</Text>
            </View>

            <FlatList
                data={workoutList}
                renderItem={({ item, index }) => (
                    <TouchableOpacity onPress={() => navigation.navigate('Workout Details', {
                        workoutTitle: item.workoutTitle,
                        exercises: item.exercises,
                        _completionStatus: item.exercises.map(exercise => {
                            let arr = [];
                            for (let i = 0; i < exercise.tableData.length; i++) {
                                arr.push(exercise.tableData[i].completed);
                            }
                            return arr;
                        }),
                        id: item.id,
                        forViewingOnly: true,
                    })}>
                        <Card>
                            <View style={styles.cardHeader}>
                                <Text style={globalStyles.titleText}>{item.workoutTitle}</Text>
                            </View>
                        </Card>
                    </TouchableOpacity>

                )}
                keyExtractor={(item, index) => item + index}
            />
        </View>

    )
}

const styles = StyleSheet.create({
    cardHeader: {
        flexDirection: 'row',
        justifyContent: 'space-between'
    },
});