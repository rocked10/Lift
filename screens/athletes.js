import React, { useState, useEffect, useRef } from 'react';
import { View, StyleSheet, FlatList, Alert, } from "react-native";
import * as Auth from '../api/auth';
import * as DB from '../api/database';
import { Button, TextInput } from "react-native-paper"
import ProfileCard from "../shared/profileCard";
import { MaterialCommunityIcons } from "@expo/vector-icons";

export default function Athletes({ navigation, route }) {
    const coachId = Auth.getCurrentUserId();

    const [coachProfile, setCoachProfile] = useState({});
    const textInputRef = useRef();

    useEffect(() => {
        DB.getUserProfile(Auth.getCurrentUserId(), setCoachProfile);
    }, []);

    const [searchQuery, setSearchQuery] = useState('')

    const handleAddAthlete = async () => {
        // Keyboard.dismiss()
        const athlete = await DB.findUserByEmail(searchQuery.toLowerCase().trim());
        if (athlete) {
            DB.addAthlete(coachId, athlete.id).then();
            Alert.alert(
                '',
                'Athlete added!',
                [
                    {
                        text: "Ok",
                    }
                ]
            );
            textInputRef.current.clear();
        } else {
            Alert.alert(
                '',
                'Athlete not found',
                [
                    {
                        text: "Ok",
                    }
                ]
            );
        }
    }

    const FlatListItemSeparator = () => {
        return (
          //Item Separator
          <View style={styles.listItemSeparatorStyle} />
        );
    };

    const handleDelete = (item) => {
        Alert.alert(
            "",
            `Remove ${item.name} from your list?`,
            [
                { text: "Cancel", onPress: () => console.log("Cancel Pressed") },
                { text: "OK", onPress: () => { DB.deleteAthlete(Auth.getCurrentUserId(), item.id).then() }}
            ]
        );
    }

    // why didn't it work when i placed the textinput and button outside??

    return (
        <View style={styles.container}>
            <TextInput placeholder="Enter Athlete's email" onChangeText={query => setSearchQuery(query)} ref={textInputRef}/>
            <Button onPress={handleAddAthlete}>add</Button>
            <FlatList
                ItemSeparatorComponent={FlatListItemSeparator}
                data={coachProfile.athletes ? Object.values(coachProfile.athletes) : null}
                renderItem={
                    ({ item }) => (
                        <ProfileCard
                            title={item.name}
                            subtitle='Athlete'
                            right={() => {
                                return (
                                    <MaterialCommunityIcons
                                        onPress={() => handleDelete(item)}
                                        style={{marginRight: 10,}}
                                        name='close'
                                        size={20}
                                    />
                                );
                            }}
                            onPress={() => {
                                navigation.navigate('View Profile', {
                                    viewUser: item.id,
                                });
                            }}
                        />
                    )}
                keyExtractor={(item, index) => index}
            />
        </View>
    );
}

const styles = StyleSheet.create({ 
    container: {  
        flex: 1,   
    },   
    item: {  
        padding: 10,
        fontSize: 16,  
        height: 40,  
    },
    listItemSeparatorStyle: {
        height: 0.5,
        width: '100%',
        backgroundColor: '#C8C8C8',
    },  
})  