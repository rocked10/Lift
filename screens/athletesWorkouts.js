import React, { useState } from 'react';
import { View, Text, FlatList, TouchableOpacity, Keyboard, StyleSheet } from "react-native";
import { globalStyles } from "../styles/global";
import { Searchbar } from 'react-native-paper';
import * as DB from "../api/database";
import Card from "../shared/card";
import ProfileCard from "../shared/profileCard";


export default function AthletesWorkouts({ navigation }) {
    const [searchQuery, setSearchQuery] = React.useState('');
    const [userFound, setUserFound] = useState('');
    const [athleteId, setAthleteId] = useState('');
    const [workouts, setWorkouts] = useState([]);
    const [displayWorkouts, setDisplayWorkouts] = useState(false);

    const onChangeSearch = (query) => {
        if (query === "") {
            setUserFound('');
            setDisplayWorkouts(false);
        } 
        setSearchQuery(query);  
    }

    const onSubmitSearch = async () => {
        Keyboard.dismiss();
        const user = await DB.findUserId(searchQuery.toLowerCase().trim());
        setUserFound(user);
        setAthleteId(user.id);
    }

    const handleConfirmation = () => {
        setDisplayWorkouts(true);
        return DB.subscribe(athleteId, setWorkouts);
    }

    const ListOfAthleteWorkouts = ({ displayWorkouts, workouts }) => {
        let workoutList = [];

        for (const workout in workouts) {
            const athleteWorkout = workouts[workout];
            workoutList.push(athleteWorkout)
        }

        if (displayWorkouts) {
            return (
                <FlatList
                    data={workoutList}
                    renderItem={({ item, index }) => (
                        <TouchableOpacity onPress={() => navigation.navigate('Workout Details', {
                                workoutTitle: item.workoutTitle,
                                exercises: item.exercises,
                                _completionStatus: item.exercises.map(exercise => {
                                    let arr = [];
                                    for (let i = 0; i < exercise.tableData.length; i++) {
                                        arr.push(exercise.tableData[i].completed);
                                    }
                                    return arr;
                                    }),
                                id: item.id,
                                forViewingOnly: true,
                        })}>
                            <Card>
                                <View style={styles.cardHeader}>
                                    <Text style={globalStyles.titleText}>{item.workoutTitle}</Text>
                                </View>
                            </Card>
                        </TouchableOpacity>

                    )}
                    keyExtractor={(item, index) => item + index}
                />
            )
        } else {
            return null
        }
    }

    const ProfileCardConditional = ({ item }) => {
        if (item) {
            return (
                <View style={{ padding: 8 }}>
                    <ProfileCard title={item} subtitle="Athlete" onPress={handleConfirmation}/>
                </View>
            );
        } else {
            return null;
        }
    }

    return (
        <View style={globalStyles.container}>
            <Searchbar
                placeholder="Search for an athlete via email..."
                value={searchQuery}
                onChangeText={onChangeSearch}
                onIconPress={onSubmitSearch}
                onSubmitEditing={onSubmitSearch}
            />

            <ProfileCardConditional item={userFound.name} />
            <ListOfAthleteWorkouts workouts={workouts} displayWorkouts={displayWorkouts} />
        </View>
    );
}

const styles = StyleSheet.create({
    cardHeader: {
        flexDirection: 'row',
        justifyContent: 'space-between'
    },
});