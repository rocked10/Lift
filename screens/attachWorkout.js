import { ListItem } from "react-native-elements";
import { FlatList, StyleSheet, Text, TouchableOpacity, View } from "react-native";
import { globalStyles } from "../styles/global";
import Card from "../shared/card";
import React, { useState, } from "react";
import * as DB from "../api/database";
import * as Auth from "../api/auth";


export default function AttachWorkout({ navigation, route }) {
    const [userId, setUserId] = useState(Auth.getCurrentUserId());
    const [workouts, setWorkouts] = useState(DB.subscribeOnce(userId));


    return (
        <View style={globalStyles.container}>
            <FlatList
                showsVerticalScrollIndicator={false}
                data={workouts ? Object.values(workouts) : null}
                renderItem={({item, index}) => {
                    if (item.exercises !== undefined) {
                        let items = item.exercises.map(item2 => {
                            return (
                                <ListItem key={Math.random()}
                                          containerStyle={{padding: 0, backgroundColor: '#F5F5F5'}}
                                >
                                    <Text style={globalStyles.cardText}>{item2.exerciseName}</Text>
                                </ListItem>
                            );
                        });

                        return (
                            <TouchableOpacity
                                onPress={() => {
                                    navigation.goBack();
                                    route.params.onSelect({
                                        workout: item,
                                    });
                                }}
                            >
                                <Card>
                                    <View style={styles.cardHeader}>
                                        <Text style={globalStyles.titleText}>{item.workoutTitle}</Text>
                                    </View>
                                    {items}
                                </Card>
                            </TouchableOpacity>
                        );
                    }
                }}
                keyExtractor={(item, index) => item + index}
            />
        </View>
    );
}

const styles = StyleSheet.create({
    cardHeader: {
        flexDirection: 'row',
        justifyContent: 'space-between',
    },
});