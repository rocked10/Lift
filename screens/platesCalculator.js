import React, { useState, } from 'react';
import { View, ScrollView, Text, Alert, TouchableOpacity } from "react-native";
import { Button, TextInput } from "react-native-paper"
import { Picker as SelectPicker } from "@react-native-picker/picker";
import Slider from '@react-native-community/slider';

export const FormField = ({ onChangeText, placeholder, val, }) => {
    return (
        <TextInput
            style={{
                // borderWidth: 1,
                borderColor: '#ddd',
                fontFamily: 'lato-regular',
                // padding: 12,
                margin: 8,
                fontSize: 18,
                borderRadius: 6,
                backgroundColor: 'white'
            }}
            onChangeText={onChangeText}
            placeholder={placeholder}
            value={val}
            keyboardType='phone-pad'
            theme={{ fonts: { regular: { fontFamily: 'lato-regular' } } }}
        />
    )
}

export const BarbellWeightSelector = ({ barbellWeight, setBarbellWeight }) => {
    return (
        <View style={{ backgroundColor: 'white', margin: 8 }}>
            <Text style={{ fontSize: 16, margin: 10, fontFamily: 'lato-bold' }} >
                Barbell Type
            </Text>
            <SelectPicker
                selectedValue={barbellWeight}
                onValueChange={(itemValue, itemIndex) =>
                    setBarbellWeight(itemValue)
                }
            >
                <SelectPicker.Item label="7' Olympic Bar (20kg)" value={20} fontFamily="lato-regular" />
                <SelectPicker.Item label="6' Olympic Bar (15kg)" value={15} fontFamily="lato-regular" />
            </SelectPicker>
        </View>
    )
}

export function generatePlatesRequired(weightToLoad, platesAvailable) {
    // to accommodate decimal weights 
    let dp = new Array(4 * weightToLoad + 4);
    let plates = new Array(4 * weightToLoad * 4);
    
    for (let i = 0; i <= 4 * weightToLoad + 3; i++) {
      if (i === 0) {
        dp[i] = 0; 
        plates[i] = [];
      }
      else {
        dp[i] = Number.MAX_VALUE; 
        plates[i] = [-1];
      }
    }
    
    for (let j = 1; j <= 4 * weightToLoad + 3; j++) {
      let plateChosen = -1; 
      let indexOfSubProb = -1;
      
      for (let k = 0; k < platesAvailable.length; k++) {
        if (platesAvailable[k] <= j / 4) {
          let subProblem = dp[j - 4 * platesAvailable[k]];
          if (subProblem !== Number.MAX_VALUE && 1 + subProblem < dp[j]) {
            dp[j] = 1 + subProblem;
            plateChosen = platesAvailable[k];
            indexOfSubProb = j - 4 * platesAvailable[k];
          }
        }
      }
      
      if (indexOfSubProb !== -1) {
        plates[j] = [... plates[indexOfSubProb], plateChosen];
      }
    }
    
    return plates[4 * weightToLoad];
}

export default function PlatesCalculator() {
    const [targetWeight, setTargetWeight] = useState(0);
    const [barbellWeight, setBarbellWeight] = useState(20);
    const [selectedPlateWeights, setSelectedPlateWeights] = useState({
        TWENTY_FIVE: [25, false, 'red'],
        TWENTY: [20, false, 'blue'],
        FIFTEEN: [15, false, 'yellow'],
        TEN: [10, false, 'green'],
        FIVE: [5, false, 'black'],
        TWO_POINT_FIVE: [2.5, false, 'black'],
        TWO: [2, false, 'black'],
        ONE_POINT_FIVE: [1.5, false, 'black'],
        ONE_POINT_TWO_FIVE: [1.25, false, 'black'],
        ONE: [1, false, 'black'],
        ZERO_POINT_FIVE: [0.5, false, 'black'],
    });

    const PlateType = ({ identifier }) => {
        const weight = selectedPlateWeights[identifier][0];
        const selected = selectedPlateWeights[identifier][1];
        let backgroundColor = selectedPlateWeights[identifier][2];
        let width = weight >= 10 ? 100 : 50;
        let height = weight >= 10 ? 100 : 50;
        let borderRadius = weight >= 10 ? 50 : 25;
        let opacity = selected ? 1 : 0.2;
        let fontColor = weight >= 10 ? 'black' : 'white'

        return (
            <TouchableOpacity
                style={{
                    opacity: opacity,
                    width: width,
                    height: height,
                    borderRadius: borderRadius,
                    backgroundColor: backgroundColor,
                    alignItems: 'center',
                    justifyContent: 'center',
                }}
                onPress={() => setSelectedPlateWeights(prev => {
                    const newSelectedPlateWeights = { ...prev };
                    newSelectedPlateWeights[identifier][1] = !newSelectedPlateWeights[identifier][1];
                    return newSelectedPlateWeights;
                })}>
                <Text style={{ fontFamily: 'lato-bold', fontSize: 18, color: fontColor }}>{weight}</Text>
            </TouchableOpacity>
        )
    }

    return (
        <ScrollView>
            <FormField onChangeText={setTargetWeight} placeholder='Target Weight' val={targetWeight} />

            <BarbellWeightSelector barbellWeight={barbellWeight} setBarbellWeight={setBarbellWeight} />
            <View style={{ margin: 10, flexDirection: 'row', flexWrap: 'wrap'}}>
                <Text style={{ fontSize: 16, margin: 10, fontFamily: 'lato-bold' }} >
                    Select the plates available to your gym!
                </Text>
                <PlateType identifier='TWENTY_FIVE' />
                <PlateType identifier='TWENTY' />
                <PlateType identifier='FIFTEEN' />
                <PlateType identifier='TEN' />
                <PlateType identifier='FIVE' />
                <PlateType identifier='TWO_POINT_FIVE' />
                <PlateType identifier='TWO' />
                <PlateType identifier='ONE_POINT_FIVE' />
                <PlateType identifier='ONE_POINT_TWO_FIVE' />
                <PlateType identifier='ONE' />
                <PlateType identifier='ZERO_POINT_FIVE' />
            </View>


            <Button
                onPress={() => {
                    const weightPerSide = (targetWeight - barbellWeight) / 2;
                    let platesAvailable = [];
                    let platesInformation = Object.values(selectedPlateWeights);
                    for (let i = 0; i < platesInformation.length; i++) {
                        if (platesInformation[i][1]) {
                            platesAvailable.push(platesInformation[i][0]);
                        }
                    }
                    const platesPerSide = generatePlatesRequired(weightPerSide, platesAvailable); 
                    if (platesPerSide[0] === -1) {
                        alert("Sorry, it's impossible.")
                    } else {
                        let str = '';
                        let plateTracker = {};
                        for (let i = 0; i < platesPerSide.length; i++) {
                            plateTracker[platesPerSide[i]] = (plateTracker[platesPerSide[i]] || 0) + 1;
                        }
                        const weightsOfPlatesChosen = Object.keys(plateTracker);
                        
                        for (let i = 0; i < weightsOfPlatesChosen.length; i++) {
                            const punctuation = i === weightsOfPlatesChosen.length - 1 ? '' : ', '
                            const weight = weightsOfPlatesChosen[i];
                            const count = plateTracker[weight];
                            const noun = count > 1 ? 'plates' : 'plate'
                            str += count + 'x' + weight + 'kg ' + noun + punctuation;
                        }
                        Alert.alert("Here's what you need on each side", str);
                    }
                }}
                style={{ marginBottom: 10 }}
            >
                <Text style={{ fontFamily: 'karla-bold', fontSize: 16 }}>Calculate!</Text>
            </Button>
        </ScrollView>
    )
}

