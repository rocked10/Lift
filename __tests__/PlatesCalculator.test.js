import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import PlatesCalculator from '../screens/platesCalculator';
import {generatePlatesRequired} from '../screens/platesCalculator';

describe('Testing Plates Calculator', () => {
    const wrapper = renderer.create(<PlatesCalculator />);

    it('Should render', () => {
        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    it('Testing plate generator function', () => {
        expect(generatePlatesRequired(200, [25, 20])).toEqual([25,25,25,25,25,25,25,25]);
        expect(generatePlatesRequired(100, [20])).toEqual([20,20,20,20,20]);
        expect(generatePlatesRequired(97.5, [20,10,5,1.5,1.25])).toEqual([1.25,1.25,5,10,20,20,20,20]);
        expect(generatePlatesRequired(91.5, [25,20,10,5,1.5])).toEqual([1.5,20,20,25,25]); 
        expect(generatePlatesRequired(87.5, [25,20,15,10,5,2.5,2])).toEqual([2.5,10,25,25,25]);
        expect(generatePlatesRequired(75, [15,10,5,2.5])).toEqual([15,15,15,15,15]);
        expect(generatePlatesRequired(72.5, [15,10,5,2.5])).toEqual([2.5,10,15,15,15,15]);
        expect(generatePlatesRequired(71, [15,10,5,2.5])).toEqual([-1]);
        expect(generatePlatesRequired(71, [15,10,5,2.5,1])).toEqual([1,10,15,15,15,15]);
        expect(generatePlatesRequired(71, [15,10,5,2.5,1])).toEqual([1,10,15,15,15,15]);
    })
});