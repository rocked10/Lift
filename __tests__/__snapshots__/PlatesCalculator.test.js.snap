// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Plates Calculator Should render 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(231, 231, 231)",
            "borderTopLeftRadius": 4,
            "borderTopRightRadius": 4,
          },
          Object {
            "backgroundColor": "white",
            "borderColor": "#ddd",
            "borderRadius": 6,
            "fontFamily": "lato-regular",
            "margin": 8,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0.26)",
            "bottom": 0,
            "height": 2,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "transform": Array [
              Object {
                "scaleY": 0.5,
              },
            ],
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "paddingBottom": 0,
              "paddingTop": 0,
            },
            Object {
              "minHeight": 64,
            },
          ]
        }
      >
        <TextInput
          allowFontScaling={true}
          editable={true}
          keyboardType="phone-pad"
          multiline={false}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder="Target Weight"
          placeholderTextColor="rgba(0, 0, 0, 0.54)"
          rejectResponderTermination={true}
          selectionColor="#6200ee"
          style={
            Array [
              Object {
                "flexGrow": 1,
                "margin": 0,
                "zIndex": 1,
              },
              Object {
                "paddingLeft": 12,
                "paddingRight": 12,
              },
              Object {
                "height": 64,
              },
              Object {
                "paddingBottom": 0,
                "paddingTop": 0,
              },
              Object {
                "color": "#000000",
                "fontFamily": "lato-regular",
                "fontSize": 18,
                "fontWeight": undefined,
                "textAlign": "left",
                "textAlignVertical": "center",
              },
              false,
              Array [
                Object {},
              ],
            ]
          }
          underlineColorAndroid="transparent"
          value={0}
        />
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "white",
          "margin": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "lato-bold",
            "fontSize": 16,
            "margin": 10,
          }
        }
      >
        Barbell Type
      </Text>
      <View>
        <RNCPicker
          items={
            Array [
              Object {
                "label": "7' Olympic Bar (20kg)",
                "testID": undefined,
                "textColor": undefined,
                "value": 20,
              },
              Object {
                "label": "6' Olympic Bar (15kg)",
                "testID": undefined,
                "textColor": undefined,
                "value": 15,
              },
            ]
          }
          onChange={[Function]}
          selectedIndex={0}
          style={
            Array [
              Object {
                "height": 216,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "flexWrap": "wrap",
          "margin": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "lato-bold",
            "fontSize": 16,
            "margin": 10,
          }
        }
      >
        Select the plates available to your gym!
      </Text>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "red",
            "borderRadius": 50,
            "height": 100,
            "justifyContent": "center",
            "opacity": 0.2,
            "width": 100,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "black",
              "fontFamily": "lato-bold",
              "fontSize": 18,
            }
          }
        >
          25
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "blue",
            "borderRadius": 50,
            "height": 100,
            "justifyContent": "center",
            "opacity": 0.2,
            "width": 100,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "black",
              "fontFamily": "lato-bold",
              "fontSize": 18,
            }
          }
        >
          20
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "yellow",
            "borderRadius": 50,
            "height": 100,
            "justifyContent": "center",
            "opacity": 0.2,
            "width": 100,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "black",
              "fontFamily": "lato-bold",
              "fontSize": 18,
            }
          }
        >
          15
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "green",
            "borderRadius": 50,
            "height": 100,
            "justifyContent": "center",
            "opacity": 0.2,
            "width": 100,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "black",
              "fontFamily": "lato-bold",
              "fontSize": 18,
            }
          }
        >
          10
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "black",
            "borderRadius": 25,
            "height": 50,
            "justifyContent": "center",
            "opacity": 0.2,
            "width": 50,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "lato-bold",
              "fontSize": 18,
            }
          }
        >
          5
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "black",
            "borderRadius": 25,
            "height": 50,
            "justifyContent": "center",
            "opacity": 0.2,
            "width": 50,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "lato-bold",
              "fontSize": 18,
            }
          }
        >
          2.5
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "black",
            "borderRadius": 25,
            "height": 50,
            "justifyContent": "center",
            "opacity": 0.2,
            "width": 50,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "lato-bold",
              "fontSize": 18,
            }
          }
        >
          2
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "black",
            "borderRadius": 25,
            "height": 50,
            "justifyContent": "center",
            "opacity": 0.2,
            "width": 50,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "lato-bold",
              "fontSize": 18,
            }
          }
        >
          1.5
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "black",
            "borderRadius": 25,
            "height": 50,
            "justifyContent": "center",
            "opacity": 0.2,
            "width": 50,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "lato-bold",
              "fontSize": 18,
            }
          }
        >
          1.25
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "black",
            "borderRadius": 25,
            "height": 50,
            "justifyContent": "center",
            "opacity": 0.2,
            "width": 50,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "lato-bold",
              "fontSize": 18,
            }
          }
        >
          1
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "black",
            "borderRadius": 25,
            "height": 50,
            "justifyContent": "center",
            "opacity": 0.2,
            "width": 50,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "lato-bold",
              "fontSize": 18,
            }
          }
        >
          0.5
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "transparent",
          "borderColor": "transparent",
          "borderRadius": 4,
          "borderStyle": "solid",
          "borderWidth": 0,
          "elevation": 0,
          "marginBottom": 10,
          "minWidth": 64,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": undefined,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 4,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "letterSpacing": 1,
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  undefined,
                  Object {
                    "textTransform": "uppercase",
                  },
                  Object {
                    "color": "#6200ee",
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  Object {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  undefined,
                ],
              ]
            }
          >
            <Text
              style={
                Object {
                  "fontFamily": "karla-bold",
                  "fontSize": 16,
                }
              }
            >
              Calculate!
            </Text>
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
